front: display conflicts on the space time chart

emersion opened the pull request:
Closes: https://github.com/OpenRailAssociation/osrd/issues/8683  
Depends on: https://github.com/OpenRailAssociation/osrd/pull/9205

emersion committed:
front: display conflicts on the space time chart

Signed-off-by: Simon Ser <contact@emersion.fr>
Closes: https://github.com/OpenRailAssociation/osrd/issues/8683

clarani commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -89,6 +94,56 @@ const SimulationResults = ({
+        },
+      ];
+    });
+  }, [conflictReqsByZone, projectedZones]);
```
Move all this code in a new hook useConflicts. You can also move the call to `timetable/id/conflicts` in this hook

emersion replied:
Oh yeah, indeed! Much better.

clarani reviewed (commented):
LGTM except the hook :)

theocrsb reviewed (approved):
LGTM and tested

flomonster reviewed (approved):
Tested

clarani commented:
LGTM, but we have to wait for this [PR](https://github.com/OpenRailAssociation/osrd-ui/pull/619)

emersion commented:
Do we? Why?

emersion committed:
front: add settings panel to space time chart

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
Updated with a settings panel to show/hide conflicts.

theocrsb reviewed (approved):
tested

clarani commented on front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/Settings.tsx:
```diff
@@ -0,0 +1,43 @@
+  onClose: () => void;
+};
+
+const Settings = ({ settings, onChange, onClose }: SettingsProps) => {
```
What about SettingsPanel ?

emersion replied:
Sure, done!

clarani commented on front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/Settings.tsx:
```diff
@@ -0,0 +1,43 @@
+
+  return (
+    <div id="settings-panel" className="flex justify-end absolute">
+      <div className="settings-panel-section">
+        <div className="settings-panel-section-title">{t('timeSpaceChartSettings.paths')}</div>
```
I'm not sure, but did you reuse classes from ui-speedspacechart ?

```suggestion
      <div>
        <div>{t('timeSpaceChartSettings.paths')}</div>
```

emersion replied:
Yes, the classes are re-used. I left a note about this in Matrix: this is far from ideal, especially since some of these CSS rules are using the HTML `id` attribute. The alternative would be to duplicate the CSS (and keep both in sync when the mockup changes). Could do that if you prefer.

On the long run, I think it'd make sense to refactor the settings panel CSS inside osrd-ui to allow for code re-use.

emersion replied:
The speedspacechart classes are no longer re/ab-used.

clarani reviewed (changes requested):
I'm ok with the 1st commit if you want to merge it without the second one.

3 remarks:
- can you make the panel less transparent ? Like the one on the GEV ?
![image](https://github.com/user-attachments/assets/73e1f0f7-e13c-4d61-b8bf-d75db6d128c3)

- Bug spotted
![image](https://github.com/user-attachments/assets/21e53a1c-db16-410b-aed7-525e1a57838f)

- Bug spotted
![image](https://github.com/user-attachments/assets/0a4f6c90-a9df-4932-af7a-9d6c2d709ecc)

emersion commented:
> can you make the panel less transparent ? Like the one on the GEV ?

Done!

> Bug spotted

Indeed, fixed.

> Bug spotted

That second one is fixed here: https://github.com/OpenRailAssociation/osrd-ui/pull/647

emersion commented:
Rebased to resolve conflicts and updated to follow this mockup:

- https://www.sketch.com/s/bcbac4ff-7f74-4f1a-9e10-08d115c47ed8/a/elyoLx2
- https://www.sketch.com/s/bcbac4ff-7f74-4f1a-9e10-08d115c47ed8/a/aMyqExr

clarani commented:
LGTM :white_check_mark:
